src/combinatorics/basic.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+
+import data.finset
+import data.fintype.basic
+
+open finset
+
+variable {α : Type*}
+variable {r : ℕ}
+
+/-!
+Basic definitions for finite sets which are useful for combinatorics.
+We define antichains, and a proposition asserting that a set is a set of r-sets.
+-/
+
+/--
+A family of sets is an antichain if no set is a subset of another. For example,
+{{1}, {4,6,7}, {2,4,5,6}} is an antichain.
+-/
+def antichain (𝒜 : finset (finset α)) : Prop := ∀ A ∈ 𝒜, ∀ B ∈ 𝒜, A ≠ B → ¬(A ⊆ B)
```

Comment:
It would be nice to add `finset.pairwise` connected both to `set.pairwise_on` and `multiset.pairwise`.

Reply:
BTW, it would be nice to define `antichain [has_le α] (s : set α) : Prop := pairwise_on s (λ x y, not (x ≤ y))`.

Reply:
I think @khanh93 might be working with a more general definition of antichain which we could use

Reply:
What definition?

Reply:
Essentially what you describe I think! That was intended more so that Ryan and Jalex are aware rather than to correct you :)

Reply:
(I was mentioned, but don't have anything to add to this discussion.)

Reply:
> BTW, it would be nice to define `antichain [has_le α] (s : set α) : Prop := pairwise_on s (λ x y, not (x ≤ y))`.

`is_antichain r s` does this now.